### ISSUE TITLE
Segmentation fault when calling ray.put on a dictionary with object keys

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#543]
Ray seg-faults when you try to put an dict containing a object instance in the key. Here's a minimal reproduction, though typically you would use some custom class instead of `object()` as the key.

```
>>> ray.register_class(object, pickle=True)
>>> ray.put({object(): object()})
[INFO] (/home/eric/Desktop/ray/src/local_scheduler/local_scheduler.cc:941) Disconnecting client on fd 12
[INFO] (/home/eric/Desktop/ray/src/plasma/plasma_store.cc:732) Disconnecting client on fd 10
[INFO] (/home/eric/Desktop/ray/src/plasma/plasma_manager.cc:1577) Disconnecting client on fd 13
zsh: segmentation fault  python3
```

**Comment by robertnishihara** [comment]
Ok, I can reproduce that, will look into it!

**Comment by robertnishihara** [comment]
Fixed by #548.
